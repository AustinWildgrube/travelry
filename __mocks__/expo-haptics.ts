module.exports = {
  ...jest.requireActual('expo-haptics'),
  impactAsync: jest.fn(),
};
