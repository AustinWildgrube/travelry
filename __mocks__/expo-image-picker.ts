module.exports = {
  __esModule: true,
  launchImageLibraryAsync: jest.fn().mockReturnValue({
    canceled: false,
    assets: [
      {
        uri: 'file://test-uri',
      },
    ],
  }),
  MediaTypeOptions: {
    All: 'all',
    Videos: 'videos',
    Images: 'images',
  },
};
