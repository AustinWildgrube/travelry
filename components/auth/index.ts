export * from './RegistrationForm';
export * from './SetupConfetti';
export * from './SetupProfileDetails';
export * from './SetupProfileImage';