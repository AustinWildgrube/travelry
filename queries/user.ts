import { Alert } from 'react-native';

import { decode } from 'base64-arraybuffer';
import { CryptoDigestAlgorithm, digestStringAsync } from 'expo-crypto';
import { type ImageResult } from 'expo-image-manipulator';

import { createAvatarsSignedUploadUrl } from '&/queries/signature';
import { supabase } from '&/services/supabase';
import { type User, type UserFollows, type UserSummary } from '&/types/types';

export const getUserAccount = async (id: string, currentUserId: string): Promise<User> => {
  // TODO: change cover photo to be anything not just a post media
  // TODO: allow for post_count to increment
  const { data, error } = await supabase
    .from('account')
    .select(
      `
      id,
      username, 
      first_name,
      last_name,
      avatar_url,
      avatar_placeholder,
      bio,
      account_stat (
        following_count, 
        followers_count, 
        trip_count
      ),
      album (
        id,
        name,
        cover,
        cover_placeholder,
        post_count
      ),
      is_following:follow!target_account_id(count)
    `,
    )
    .eq('id', id)
    .eq('follow.account_id', currentUserId)
    .eq('follow.target_account_id', id)
    .order('name', { foreignTable: 'album', ascending: true })
    .single();

  if (error) throw new Error(`getUser(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const user = data as User;
  const userCount = user.is_following as { count: number }[];
  user.is_following = userCount[0].count > 0;

  console.log(user);

  return user;
};

export const getUserAccountSearch = async (searchText: string): Promise<UserSummary[]> => {
  const { data: conversationRecipients, error } = await supabase
    .from('account')
    .select(
      `
      id,
      username,
      first_name,
      last_name,
      avatar_url,
      avatar_placeholder
    `,
    )
    .or(`username.ilike.%${searchText}%,first_name.ilike.%${searchText}%,last_name.ilike.%${searchText}%`);

  if (error) throw new Error(`getUserAccountSearch(${error.code}): ${error.message}`);

  return conversationRecipients as UserSummary[];
};

export const updateUserAccount = async (id: string, updates: Partial<User>): Promise<void> => {
  const { error } = await supabase.from('account').update(updates).eq('id', id);
  if (error) throw new Error(`updateUserAccount(${error.code}): ${error.message}`);
};

export const uploadUserAvatar = async (userId: string, avatarImage: ImageResult): Promise<void> => {
  if (!avatarImage.base64) return;

  const hash = await digestStringAsync(CryptoDigestAlgorithm.SHA256, avatarImage.uri);
  const filename = `${hash}.jpg`;

  const signedUrl = await createAvatarsSignedUploadUrl(filename);
  const { error } = await supabase.storage.from('avatars').uploadToSignedUrl(filename, signedUrl, decode(avatarImage.base64), {
    upsert: true,
    contentType: 'image/jpeg',
  });

  if (error) {
    if (error.statusCode === 413) Alert.alert('Image is too large', 'Please select an image under 2MB.');
    throw new Error(`uploadUserAvatarOne(${error.statusCode}): ${error.message}`);
  }

  const { error: errorTwo } = await supabase.from('account').update({ avatar_url: filename }).eq('id', userId);
  if (errorTwo) throw new Error(`uploadUserAvatarTwo(${errorTwo.code}): ${errorTwo.message}`);
};

export const followUser = async (id: string, targetId: string): Promise<void> => {
  let { error } = await supabase.from('follow').insert({ account_id: id, target_account_id: targetId });
  if (error) throw new Error(`followUser(${error.code}): ${error.message}`);
};

export const unfollowUser = async (id: string, targetId: string): Promise<void> => {
  let { error } = await supabase.from('follow').delete().eq('account_id', id).eq('target_account_id', targetId);
  if (error) throw new Error(`unfollowUser(${error.code}): ${error.message}`);
};

export const getUserFollowers = async (id: string): Promise<UserFollows[]> => {
  const { data, error } = await supabase
    .from('follow')
    .select(
      `
        id,
        account!account_id (
          id,
          username,
          first_name,
          last_name,
          avatar_url,
          avatar_placeholder
        ),
        is_following:account!target_account_id(count)
      `,
    )
    .eq('target_account_id', id);

  if (error) throw new Error(`getUserFollowers(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const followers = data as UserFollows[];
  for (let i = 0; i < followers.length; i++) {
    if (!followers[i].is_following) {
      followers[i].is_following = false;
    } else {
      // @ts-ignore
      followers[i].is_following = followers[i].is_following.count > 0;
    }
  }

  return followers as UserFollows[];
};

export const getUserFollowing = async (id: string): Promise<UserFollows[]> => {
  const { data, error } = await supabase
    .from('follow')
    .select(
      `
        id,
        account!target_account_id (
          id,
          username,
          first_name,
          last_name,
          avatar_url,
          avatar_placeholder
        ),
        is_following:account!target_account_id(count)
      `,
    )
    .eq('account_id', id);

  if (error) throw new Error(`getUserFollowing(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const followers = data as UserFollows[];
  for (let i = 0; i < followers.length; i++) {
    if (!followers[i].is_following) {
      followers[i].is_following = false;
    } else {
      // @ts-ignore
      followers[i].is_following = followers[i].is_following.count > 0;
    }
  }

  return data as UserFollows[];
};
