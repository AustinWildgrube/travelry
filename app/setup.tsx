import { Fragment, useState } from 'react';

import { Link } from 'expo-router';
import { SafeAreaView } from 'react-native-safe-area-context';

import { Confetti, SetupProfileDetails, SetupProfileImage } from '&/components/auth';
import { Button, Heading, Text } from '&/components/core';
import { CONFETTI_FALL_SPEED, CONFETTI_ITEM_HEIGHT, CONFETTI_ITEM_WIDTH } from '&/utils/constants';
import { makeStyles } from '&/utils/makeStyles';

export default function Setup(): JSX.Element {
  const styles = useStyles();
  const [step, setStep] = useState(0);

  return (
    <SafeAreaView style={styles.setup}>
      <Heading>Setup your profile</Heading>

      {step === 0 ? (
        <Fragment>
          <Text style={styles.setupSlogan}>Upload an image - it's worth a thousand words</Text>
          <SetupProfileImage setStep={setStep} />

          <Button onPress={() => setStep(1)} variant="link" style={styles.skipLink}>
            <Button.Text>Skip</Button.Text>
          </Button>

          <Confetti
            fallSpeed={CONFETTI_FALL_SPEED}
            flipSpeed={3}
            horizSpeed={50}
            itemColors={['#84a98c', '#ffb703', '#023047']}
            itemDimensions={{ height: CONFETTI_ITEM_HEIGHT, width: CONFETTI_ITEM_WIDTH }}
            itemTintStrength={0.8}
            numItems={100}
          />
        </Fragment>
      ) : (
        <Fragment>
          <Text style={styles.setupSlogan}>Tell the world a little about yourself</Text>
          <SetupProfileDetails />

          <Link href="/feed" asChild>
            <Button variant="link" style={styles.skipLink}>
              <Button.Text>Skip</Button.Text>
            </Button>
          </Link>
        </Fragment>
      )}
    </SafeAreaView>
  );
}

const useStyles = makeStyles(theme => ({
  setup: {
    flex: 1,
    padding: theme.space['4'],
  },
  setupSlogan: {
    marginTop: theme.space['1'],
    marginBottom: theme.space['6'],
  },
  skipLink: {
    marginTop: theme.space['3'],
    zIndex: -1,
  },
}));
