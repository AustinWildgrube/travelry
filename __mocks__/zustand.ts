import { act } from '@testing-library/react-native';
import { StateCreator } from 'zustand';

import { useUserStore } from '&/stores/user';
import { type User } from '&/types/types';

const { create: actualCreate } = jest.requireActual('zustand');
const storeResetFns = new Set<() => void>();

export const mockUser: User = {
  id: '1',
  username: '',
  first_name: '',
  last_name: '',
  avatar_url: '',
  avatar_placeholder: '',
  is_following: false,
  bio: '',
  account_stat: {
    followers_count: 0,
    following_count: 0,
    trip_count: 0,
  },
  album: [
    {
      id: '',
      name: '',
      cover: '',
      cover_placeholder: '',
      post_count: 0,
    },
  ],
};

export const create = <S>(createState: StateCreator<S>) => {
  const store = actualCreate(createState);
  const initialState = store.getState();

  storeResetFns.add(() => {
    store.setState(initialState, true);
  });

  return store;
};

beforeEach(async () => {
  await act(() => {
    storeResetFns.forEach(resetFn => {
      resetFn();
    });

    useUserStore.setState({ setUser: undefined, updateUserStats: undefined, user: mockUser });
  });
});
