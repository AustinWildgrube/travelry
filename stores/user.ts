import { create } from 'zustand';

import { type User, type UserStat } from '&/types/types';

type UserState = {
  user: User | null;
  setUser: (user: User) => void;
  updateUserStats: (stats: UserStat) => void;
};

export const useUserStore = create<UserState>(set => ({
  user: null,
  setUser: (user: User) => {
    set({ user });
  },
  updateUserStats: (stats: UserStat) => {
    set(state => ({
      user: state.user
        ? {
            ...state.user,
            account_stat: {
              ...state.user.account_stat,
              ...stats,
            },
          }
        : null,
    }));
  },
}));
