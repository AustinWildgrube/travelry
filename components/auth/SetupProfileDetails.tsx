import { KeyboardAvoidingView, ScrollView } from 'react-native';

import { zodResolver } from '@hookform/resolvers/zod';
import { useRouter } from 'expo-router';
import { Controller, useForm } from 'react-hook-form';
import Animated, { FadeInRight } from 'react-native-reanimated';
import { z } from 'zod';

import { Button, FormControl, Icon, Input, Textarea } from '&/components/core';
import { WarningIcon } from '&/components/core/Icons/Icons/Warning';
import { LoadingIndicator } from '&/components/shared/LoadingIndicator';
import { updateUserAccount } from '&/queries/user';
import { useUserStore } from '&/stores/user';
import { makeStyles } from '&/utils/makeStyles';

const SetupFormSchema = z.object({
  first_name: z.string().optional(),
  last_name: z.string().optional(),
  location: z.string().optional(),
  bio: z.string().optional(),
});

type SetupForm = z.infer<typeof SetupFormSchema>;

export function SetupProfileDetails(): JSX.Element {
  const styles = useStyles();
  const router = useRouter();

  const user = useUserStore(state => state.user);
  if (!user) return <LoadingIndicator />;

  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<SetupForm>({
    resolver: zodResolver(SetupFormSchema),
  });

  const submitSetupForm = async (data: SetupForm): Promise<void> => {
    await updateUserAccount(user.id, data);
    router.replace('/(tabs)/feed');
  };

  return (
    <Animated.View entering={FadeInRight.duration(500)} style={styles.profileDetails}>
      <ScrollView>
        <KeyboardAvoidingView behavior="position">
          <Controller
            name="first_name"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <FormControl isInvalid={!!errors.first_name} style={styles.formControl}>
                <FormControl.Label>
                  <FormControl.Label.Text>First Name</FormControl.Label.Text>
                </FormControl.Label>

                <Input>
                  <Input.Input placeholder="First Name" onChangeText={onChange} onBlur={onBlur} value={value} />
                </Input>

                <FormControl.Error>
                  <FormControl.Error.Icon>
                    <Icon as={WarningIcon} style={styles.formControlIcon} />
                  </FormControl.Error.Icon>

                  <FormControl.Error.Text>{errors.first_name?.message}</FormControl.Error.Text>
                </FormControl.Error>
              </FormControl>
            )}
          />

          <Controller
            name="last_name"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <FormControl isInvalid={!!errors.last_name} style={styles.formControl}>
                <FormControl.Label>
                  <FormControl.Label.Text>Last Name</FormControl.Label.Text>
                </FormControl.Label>

                <Input>
                  <Input.Input placeholder="Last Name" onChangeText={onChange} onBlur={onBlur} value={value} />
                </Input>

                <FormControl.Error>
                  <FormControl.Error.Icon>
                    <Icon as={WarningIcon} style={styles.formControlIcon} />
                  </FormControl.Error.Icon>

                  <FormControl.Error.Text>{errors.last_name?.message}</FormControl.Error.Text>
                </FormControl.Error>
              </FormControl>
            )}
          />

          <Controller
            name="location"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <FormControl isInvalid={!!errors.location} style={styles.formControl}>
                <FormControl.Label>
                  <FormControl.Label.Text>Location</FormControl.Label.Text>
                </FormControl.Label>

                <Input>
                  <Input.Input placeholder="Location" onChangeText={onChange} onBlur={onBlur} value={value} />
                </Input>

                <FormControl.Error>
                  <FormControl.Error.Icon>
                    <Icon as={WarningIcon} style={styles.formControlIcon} />
                  </FormControl.Error.Icon>

                  <FormControl.Error.Text>{errors.location?.message}</FormControl.Error.Text>
                </FormControl.Error>
              </FormControl>
            )}
          />

          <Controller
            name="bio"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <FormControl isInvalid={!!errors.bio} style={styles.formControl}>
                <FormControl.Label>
                  <FormControl.Label.Text>Bio</FormControl.Label.Text>
                </FormControl.Label>

                <Textarea>
                  <Textarea.Input placeholder="Tell the world a little bit about yourself" onChangeText={onChange} onBlur={onBlur} value={value} />
                </Textarea>

                <FormControl.Error>
                  <FormControl.Error.Icon>
                    <Icon as={WarningIcon} style={styles.formControlIcon} />
                  </FormControl.Error.Icon>

                  <FormControl.Error.Text>{errors.bio?.message}</FormControl.Error.Text>
                </FormControl.Error>
              </FormControl>
            )}
          />
        </KeyboardAvoidingView>
      </ScrollView>

      <Button onPress={handleSubmit(submitSetupForm)} style={styles.submitButton}>
        <Button.Text>Finish</Button.Text>
      </Button>
    </Animated.View>
  );
}

const useStyles = makeStyles(theme => ({
  profileDetails: {
    flex: 1,
    justifyContent: 'center',
  },
  formControl: {
    marginBottom: theme.space['4'],
  },
  formControlIcon: {
    color: theme.colors['red500'],
    height: theme.space['3'],
    width: theme.space['3'],
  },
  submitButton: {
    marginTop: theme.space['8'],
  },
}));
