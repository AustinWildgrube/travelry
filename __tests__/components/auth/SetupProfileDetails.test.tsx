import { fireEvent, screen, waitFor } from '@testing-library/react-native';

import { SetupProfileDetails } from '&/components/auth/SetupProfileDetails';
import { updateUserAccount } from '&/queries/user';

import { mockUser } from '&/__mocks__/zustand';
import { render } from '&/jest.setup';

jest.mock('&/queries/user', () => ({
  updateUserAccount: jest.fn(),
}));

describe('SetupProfileDetails', () => {
  beforeEach(() => {
    render(<SetupProfileDetails />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    screen.toJSON();
    expect(screen).toMatchSnapshot();
  });

  it('should render all input fields', () => {
    const firstNameInput = screen.getByPlaceholderText(/First Name/i);
    const lastNameInput = screen.getByPlaceholderText(/Last Name/i);
    const locationInput = screen.getByPlaceholderText(/Location/i);
    const bioInput = screen.getByPlaceholderText(/Tell the world a bit about yourself/i);

    expect(firstNameInput).toBeDefined();
    expect(lastNameInput).toBeDefined();
    expect(locationInput).toBeDefined();
    expect(bioInput).toBeDefined();
  });

  it('should submit the form when all inputs are valid', async () => {
    fireEvent.changeText(screen.getByPlaceholderText(/First Name/i), 'John');
    fireEvent.changeText(screen.getByPlaceholderText(/Last Name/i), 'Doe');
    fireEvent.changeText(screen.getByPlaceholderText(/Location/i), 'New York');
    fireEvent.changeText(screen.getByPlaceholderText(/Tell the world a bit about yourself/i), 'Hello, world!');

    fireEvent.press(screen.getByRole('button', { name: /Finish/i }));

    await waitFor(() =>
      expect(updateUserAccount).toHaveBeenCalledWith(mockUser.id, {
        first_name: 'John',
        last_name: 'Doe',
        location: 'New York',
        bio: 'Hello, world!',
      }),
    );
  });
});
