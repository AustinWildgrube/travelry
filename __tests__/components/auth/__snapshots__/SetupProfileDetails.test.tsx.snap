// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupProfileDetails should match the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "zIndex": 1,
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      entering={
        FadeInRight {
          "build": [Function],
          "durationV": 500,
          "randomizeDelay": false,
        }
      }
    >
      <RCTScrollView>
        <View>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                {
                  "bottom": 0,
                }
              }
            >
              <View
                style={
                  {
                    "marginBottom": 12,
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#242424",
                      "fontFamily": "Inter_600SemiBold",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginLeft": 4,
                    }
                  }
                >
                  First Name
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderBottomWidth": 2,
                      "borderColor": "#d0d5dd",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 2,
                      "elevation": 2,
                      "flexDirection": "row",
                      "marginTop": 4,
                      "shadowColor": "rgba(16, 24, 40, .04)",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 2,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="First Name"
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "400",
                        "minHeight": 40,
                        "padding": 12,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#8d9494",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "marginBottom": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "marginBottom": 12,
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#242424",
                      "fontFamily": "Inter_600SemiBold",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginLeft": 4,
                    }
                  }
                >
                  Last Name
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderBottomWidth": 2,
                      "borderColor": "#d0d5dd",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 2,
                      "elevation": 2,
                      "flexDirection": "row",
                      "marginTop": 4,
                      "shadowColor": "rgba(16, 24, 40, .04)",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 2,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Last Name"
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "400",
                        "minHeight": 40,
                        "padding": 12,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#8d9494",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "marginBottom": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "marginBottom": 12,
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#242424",
                      "fontFamily": "Inter_600SemiBold",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginLeft": 4,
                    }
                  }
                >
                  Location
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderBottomWidth": 2,
                      "borderColor": "#d0d5dd",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 2,
                      "elevation": 2,
                      "flexDirection": "row",
                      "marginTop": 4,
                      "shadowColor": "rgba(16, 24, 40, .04)",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 2,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Location"
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 16,
                        "fontWeight": "400",
                        "minHeight": 40,
                        "padding": 12,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#8d9494",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "marginBottom": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "marginBottom": 12,
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#242424",
                      "fontFamily": "Inter_600SemiBold",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginLeft": 4,
                    }
                  }
                >
                  Bio
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderBottomWidth": 2,
                      "borderColor": "#d0d5dd",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 2,
                      "elevation": 2,
                      "flexDirection": "row",
                      "marginTop": 4,
                      "shadowColor": "rgba(16, 24, 40, .04)",
                      "shadowOffset": {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 2,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Tell the world a bit about yourself"
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "height": 200,
                        "minHeight": 40,
                        "padding": 12,
                        "paddingTop": 12,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#8d9494",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "marginBottom": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 8,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "paddingTop": 24,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#84a98c",
                "borderBottomWidth": 4,
                "borderColor": "#6A8770",
                "borderRadius": 8,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 12,
                "paddingHorizontal": 14,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Finish
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
